python/src/trezorlib/cli/firmware.py

```diff
@@ -635,15 +663,57 @@ def update(
             model=client.model,
         )
 
-    if not raw:
-        firmware_data = extract_embedded_fw(
-            firmware_data=firmware_data,
-            bootloader_onev2=_is_bootloader_onev2(client),
-        )
+        if not raw:
+            firmware_data = extract_embedded_fw(
+                firmware_data=firmware_data,
+                bootloader_onev2=_is_bootloader_onev2(client),
+            )
 
     if dry_run:
         click.echo("Dry run. Not uploading firmware to device.")
-    else:
+        return
+
+    print(ilu)
+
+    if ilu is False:
+        strict_upgrade = _is_strict_update(client, firmware_data)
+        ilu_supported = _is_ilu_supported(client)
+
+        if ilu is True:
+            if client.features.bootloader_mode:
+                click.echo(
+                    "Interaction-less update can not be started from bootloader mode."
+                )
+                exit(1)
+            elif not ilu_supported:
+                click.echo(
+                    "Interaction-less upgrade is not supported by the current firmware version."
+                )
+                exit(1)
+            elif not strict_upgrade:
```

Comment:
maybe we could ignore this if skip_check is set

Reply:
Not relevant already. As @matejcik suggested, I've removed the --ilu/--no-ilu arguments.